src/aatoolbox/datasources/chirps/chirps.py

```diff
@@ -0,0 +1,631 @@
+                "indicate the range for which CHIRPS data are "
+                "currently available."
+            )
+
+        if self._frequency == "monthly":
+            date_list = [
+                (d.split("-")[0], d.split("-")[1])
+                for d in pd.date_range(
+                    f"{start_year}-{start_month}",
+                    f"{end_year}-{end_month}",
+                    freq="MS",
+                )
+                .strftime("%Y-%m")
+                .tolist()
+            ]
+        elif self._frequency == "daily":
+            date_list = [
+                (d.split("-")[0], d.split("-")[1], d.split("-")[2])
+                for d in pd.date_range(
+                    f"{start_year}-{start_month}-{start_day}",
+                    f"{end_year}-{end_month}-{end_day}",
+                    freq="D",
+                )
+                .strftime("%Y-%m-%d")
+                .tolist()
+            ]
+
+        # Create a sentence containing information on the downloaded data
+        if len(date_list) == 0:
+            sentence_to_print = (
+                "No data will be downloaded. "
+                "There is no {self._frequency} available within the "
+                "chosen range of dates."
+            )
+        else:
+            sentence_to_print = (
+                f"{self._frequency.capitalize()} "
+                "data will be downloaded, starting from "
+                f"{'-'.join(date_list[0])} to {'-'.join(date_list[-1])}."
+            )
+
+        return date_list, sentence_to_print
+
+    def _get_file_name(
+        self,
+        year: Optional[Union[int, str]],
+        month: Optional[Union[int, str]],
+        day: Optional[Union[int, str]],
+    ) -> str:
+        # Set wildcard for year, to get general filename pattern
+        # used to find all files with pattern in folder
+        if year is None:
+            year = "*"
+        if month is None:
+            month = "*"
+        else:
+            if len(f"{month}") == 1:
+                month = f"0{month}"
+        if day is None:
+            day = "*"
+        else:
+            if len(f"{day}") == 1:
+                day = f"0{day}"
+
+        file_name_base = (
+            f"{self._country_config.iso3}_chirps_"
+            f"{self._frequency}_{year}_{month}_"
+        )
+        if self._frequency == "daily":
+            file_name_base += f"{day}_"
+        file_name = (
+            f"{file_name_base}"
+            f"r{self._resolution}"
+            f"_{self._geobb.get_filename_repr(p=0)}.nc"
+        )
+        return file_name
+
+    def _get_raw_path(
+        self, year: Optional[str], month: Optional[str], day: Optional[str]
+    ) -> Path:
+        return self._raw_base_dir / self._get_file_name(
+            year=year, month=month, day=day
+        )
+
+    def _get_processed_path(self, raw_path: Path) -> Path:
+        return self._processed_base_dir / raw_path.parts[-1]
```

Comment:
I'm not sure whether this would make things easy as, in order to get the file name, you need to specify year, month and day, which you would need to get anyway from the raw_path.

Reply:
I'm not sure I understood what you mean. @turnerm, any idea about Tinka's comment?